sheetScript.gs

```diff
@@ -117,11 +141,37 @@ function updatePlaylists(sheet) {
     deletePlaylistItems(playlistId, deleteBeforeTimestamp);
     
   }
-  if (!debugFlag_dontUpdateTimestamp && !errorflag) sheet.getRange(reservedTimestampCell).setValue(ISODateString(new Date())); // Update timestamp
+  
+  // Prints logs to Debug sheet
+  var debugSheet = spreadsheet.getSheetByName("Debug")
+  if (!debugSheet) debugSheet = spreadsheet.insertSheet("Debug")
+  var newLogs = Logger.getLog().split("\n").slice(0, -1).map(function(log) {return log.split(" INFO: ")})
+  if (newLogs.length > 0) debugSheet.clear().getRange(1, 1, newLogs.length, 2).setValues(newLogs)
+
+  if (errorflag) throw new Error(errorCount+" videos were not added to playlist correctly, please check Debug sheet. Timestamp has not been updated.")
+  if (!debugFlag_dontUpdateTimestamp) sheet.getRange(reservedTimestampCell).setValue(ISODateString(new Date())); // Update timestamp
 }
 
 function getVideoIds(channelId, lastTimestamp) {
   var videoIds = [];
+  
+  try {
+    // Check Channel validity
+    var results = YouTube.Channels.list('id', {
+      id: channelId
+    });
+    if (!results || !results.items) {
+      Logger.log("YouTube channel search returned invalid response for channel with id "+channelId)
+      return []
+    } else if (results.items.length === 0) {
+      Logger.log("Cannot find channel with id "+channelId)
+      return []
+    }
+  } catch (e) {
+    Logger.log("Cannot lookup channel on YouTube, ERROR: " + e.message);
+    return [];
+  }
+
```

Comment:
Does this double the quota usage?

Reply:
Moved so that it only checks if no videos are found. Checking this way is the only way to see if channel/playlist is valid since both channel doesn't exist and channel has no new videos produce the same result.